src/shell/server.cpp

```diff
@@ -486,15 +487,23 @@ server::cmd_res server::handle_sync(server::cmd_req const & req) {
     } else {
         new_content = load_module(new_file_name, /* can_use_olean */ false)->m_contents;
     }
-
-    auto mtime = time(nullptr);
+    unsigned new_hash;
+    {
+        std::string tmp = new_content;
+        remove_cr(tmp);
+        // TODO(Vtec234): by how much is hashing on sync going to slow the server down?
```

Comment:
I just measured it on my laptop, and hashing all of mathlib takes around 1s the first time, then .1s with a hot disk cache, so this should not be an issue. Note that the hash on olean files serves a different purpose, since it's derived from the object code rather than the .lean source file.